contracts/reward/RewardsDistributor.sol

```diff
@@ -0,0 +1,176 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.7.6;
+pragma abicoder v2;
+
+import {ReentrancyGuard} from '@openzeppelin/contracts/utils/ReentrancyGuard.sol';
+import {SafeMath} from '@openzeppelin/contracts/math/SafeMath.sol';
+import {SafeERC20} from '@openzeppelin/contracts/token/ERC20/SafeERC20.sol';
+import {MerkleProof} from '@openzeppelin/contracts/cryptography/MerkleProof.sol';
+import {Utils} from '@kyber.network/utils-sc/contracts/Utils.sol';
+import {PermissionAdmin} from '@kyber.network/utils-sc/contracts/PermissionAdmin.sol';
+import {IPool} from '../interfaces/liquidation/IPool.sol';
+import {IERC20Ext, IRewardsDistributor} from '../interfaces/rewardDistribution/IRewardsDistributor.sol';
+
+
+contract RewardsDistributor is IRewardsDistributor, PermissionAdmin, ReentrancyGuard, Utils {
+  using SafeERC20 for IERC20Ext;
+  using SafeMath for uint256;
+
+  struct MerkleData {
+    uint256 cycle;
+    bytes32 root;
+    string contentHash;
+  }
+
+  IPool public treasuryPool;
+  MerkleData private merkleData;
+  // wallet => token => claimedAmount
+  mapping(address => mapping(IERC20Ext => uint256)) public claimedAmounts;
+
+  event TreasuryPoolSet(IPool indexed treasuryPool);
+  event Claimed(
```

Comment:
may be move the event to Interface instead

Reply:
updated